torch/csrc/jit/passes/graph_rewrite_helper.cpp

```diff
@@ -66,17 +73,40 @@ void replaceConvolutionWithConv2d(std::shared_ptr<Graph>& graph) {
     auto output_padding_value =
         getIValue("output_padding", match_vmap, vmap).value().toIntList();
 
-    if (!transposed_value && !benchmark_value && !deterministic_value &&
-        cudnn_enabled_value && (output_padding_value[0] == 0) &&
-        (output_padding_value[1] == 0)) {
-      return true;
+    if (output_padding_value.size() != 1) {
+      return false;
     }
-    return false;
+    return !transposed_value && !benchmark_value && !deterministic_value &&
+        cudnn_enabled_value && (output_padding_value[0] == 0);
+  };
+  auto filter_conv2d = [](const Match& match,
+                          const std::unordered_map<std::string, Value*>& vmap) {
+    const auto& match_vmap = match.values_map;
+    auto transposed_value =
+        getIValue("transposed", match_vmap, vmap).value().toBool();
+    auto benchmark_value =
+        getIValue("benchmark", match_vmap, vmap).value().toBool();
+    auto deterministic_value =
+        getIValue("deterministic", match_vmap, vmap).value().toBool();
+    auto cudnn_enabled_value =
+        getIValue("cudnn_enabled", match_vmap, vmap).value().toBool();
+    auto output_padding_value =
+        getIValue("output_padding", match_vmap, vmap).value().toIntList();
+
+    if (output_padding_value.size() != 2) {
+      return false;
+    }
+    return !transposed_value && !benchmark_value && !deterministic_value &&
+        cudnn_enabled_value && (output_padding_value[0] == 0) &&
+        (output_padding_value[1] == 0);
   };
```

Comment:
there are some duplicated code here, could you extract the common parts?

Reply:
Sure